.github/workflows/ad_hoc_dev_deploy.yml

```diff
@@ -0,0 +1,54 @@
+name: Deploy branch to dev
+run-name: Deploy branch '${{github.ref_name}}' to dev
+
+permissions:
+  contents: read  # This is required for actions/checkout
+  id-token: write  # This is required for authenticating with aws
+
+on:
+  workflow_dispatch:
+    inputs:
+        run_e2e_tests:
+            required: false
+            default: true
+            type: boolean
+            description: Run e2e tests
+  schedule:
+  - cron:  '0 2 * * 1'
+jobs:
+  paketo_build:
+    name: Build
+    uses: ./.github/workflows/package.yml
+    secrets:
+      AWS_ACCOUNT: ${{ secrets.AWS_ACCOUNT }}
+      CI_AWS_ACCOUNT: ${{ secrets.CI_AWS_ACCOUNT }}
+      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      TEMP_SLACK_CHANNEL_ID: ${{ secrets.TEMP_SLACK_CHANNEL_ID }}
+
+  run_e2e_tests_docker_compose:
+    name: Run E2E tests against docker compose
+    uses: ./.github/workflows/run_e2e_tests_docker_compose.yml
+    secrets:
+      PULLPREVIEW_GOVUK_NOTIFY_API_KEY: ${{ secrets.PULLPREVIEW_GOVUK_NOTIFY_API_KEY }}
+      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      TEMP_SLACK_CHANNEL_ID: ${{ secrets.TEMP_SLACK_CHANNEL_ID }}
```

Comment:
I think we might only want this for the scheduled deployment? I can imagine wanting to deploy a branch to dev where you have broken the e2e tests/not updated the e2e tests yet?

Reply:
Yes, fair point, I'll update this to also be conditional on the e2e input.